File: ___TB1/src/S-1b/implementation/Snake.cpp
// Snake.cpp

#include "../header/Snake.h"
#include <iostream>
#include <thread>
#include <chrono>
using namespace std;

Snake::Snake(const Point& P) : Hewan(SNAKE_ID, SNAKE_MAXAGE)
{
    age = 0;
	status = 1;
	power = SNAKE_BASEPOWER;
	pos = P;
}

File: TubesMakhluk/src/makhluk/Snake.java
package makhluk;

import point.Point;

/** Kelas turunan dari Hewan yang merepresentasikan seekor ular
 *  @author     Micky Yudi Utama
 *  @version	1.0
 */
public class Snake extends Hewan {
    /** Atribut ID ular.
    */    
    private static final char SNAKE_ID = 'U';
    /** Atribut umur maksimal ular.
    */    
    private static final int SNAKE_MAXAGE = 150;
    /** Atribut ID ular.
    */    
    private static final int SNAKE_BASEPOWER = 5;
    /** Menciptakan seekor ular di koordinat p.
     *  @param P koordinat penciptaan ular
     */
    public Snake(final Point p) {
        super(SNAKE_ID, SNAKE_MAXAGE);
        age = 0;
        status = 1;
        power = SNAKE_BASEPOWER;
        pos = p;
    }
}
